lib/Common/Memory/PageAllocator.cpp

```diff
@@ -124,6 +124,8 @@ SegmentBase<T>::Initialize(DWORD allocFlags, bool excludeGuardPages)
         return false;
     }
 
+    Assert( ((ULONG_PTR)this->address % (64 * 1024)) == 0 );
```

Comment:
Instead of 64 * 1024, can you use AutoSystemInfo::Data.dwAllocationGranularity here? Also, in SysInfo.cpp, after the call to GetSystemInfo in AutoSystemInfo's constructor, it's worth adding a similar assert (that dwAllocationGranularity is 64k)